src/js/AccessibilityUtils.js

```diff
@@ -1041,30 +1041,113 @@ axs.utils.getQuerySelectorText = function(obj) {
+    if (!id)
         return null;
-    var propertyType = property.valueType;
-    if (propertyType === 'idref_list' || propertyType === 'idref') {
-        id = id.replace(/'/g, "\\'");
-        var referrerQuery = "[" + attributeName + "~='" + id + "']";
-        return element.ownerDocument.querySelectorAll(referrerQuery);
+    id = id.replace(/'/g, "\\'");  // make it safe to use in a selector
+
+    if (opt_attributeName) {
+        var propertyKey = opt_attributeName.replace(/^aria-/, '');
+        var referrerQuery = propertyToSelector(propertyKey);
+        if (referrerQuery) {
+            return element.ownerDocument.querySelectorAll(referrerQuery);
+        }
+    } else {
+        var selectors = [];
+        for (var propertyKey in axs.constants.ARIA_PROPERTIES) {
+            var referrerQuery = propertyToSelector(propertyKey);
+            if (referrerQuery) {
+                selectors.push(referrerQuery);
+            }
+        }
+        return element.ownerDocument.querySelectorAll(selectors.join(','));
     }
     return null;
 };
 
 /**
+ * Gets elements that refer to this element in an HTML attribute that takes an ID reference list or
+ * single ID reference.
+ * @param {Element} element a potential referent.
+ * @return {NodeList} The elements that refer to this element.
+ */
+axs.utils.getHtmlIdReferrers = function(element) {
+    if (!element)
+        return null;
+    var id = element.id;
+    if (!id)
+        return null;
+    id = id.replace(/'/g, "\\'");  // make it safe to use in a selector
+    var selectorTemplates = [
+        '[contextmenu=\'{id}\']',
+        '[itemref~=\'{id}\']',
+        'button[form=\'{id}\']',
+        'button[menu=\'{id}\']',
+        'fieldset[form=\'{id}\']',
+        'input[form=\'{id}\']',
+        'input[list=\'{id}\']',
+        'keygen[form=\'{id}\']',
+        'label[for=\'{id}\']',
+        'label[form=\'{id}\']',
+        'menuitem[command=\'{id}\']',
+        'object[form=\'{id}\']',
+        'output[for~=\'{id}\']',
+        'output[form=\'{id}\']',
+        'select[form=\'{id}\']',
+        'td[headers~=\'{id}\']',
+        'textarea[form=\'{id}\']',
+        'tr[headers~=\'{id}\']'];
+    var selectors = selectorTemplates.map(function(selector) {
+        return selector.replace('\{id\}', id);
+    });
+    return element.ownerDocument.querySelectorAll(selectors.join(','));
+};
+
+/**
+ * Gets elements that refer to this element in an attribute that takes an ID reference list or
+ * single ID reference.
+ * @param {Element} element a potential referent.
+ * @return {Array<Element>} The elements that refer to this element.
+ */
+axs.utils.getIdReferrers = function(element) {
+    var result = [];
+    var addReferrers = function() {
+        for (var i = 0; i < referrers.length; i++) {
+            result.push(referrers[i]);
+        }
+    };
+    var referrers = axs.utils.getHtmlIdReferrers(element);
+    if (referrers) {
+        addReferrers();
```

Comment:
Love it alice, very smart.
Unfortunately PhantomJS (2.0.0) can't handle it:
`'[object NodeList]' is not a valid argument for 'Function.prototype.apply' (evaluating 'Array.prototype.push.apply')`

We'd have to convert the NodeList to an Array first which somewhat defeats the purpose of what you were going for which was a nice one-liner instead of needing a helper function.

I could change it anyway if you prefer that style, including the conversion to array.